### PR TITLE
refactor: refactor create state table code + fix incorrect arguments

### PR DESCRIPTION
**Pull request by BowenXiao1999**
## What's changed and what's your intention?

It was bug introduced by incorrect arguments (My bad)... 

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/stream/src/executor/aggregation/mod.rs [hunk 1]
```diff
@@ -352,8 +353,8 @@ pub fn generate_agg_schema(
 
 /// Infer column desc for state table.
 /// The column desc layout is
-/// [ `group_key` (only for hash agg) / `sort_key` (only for simple agg) / `value`(the agg call
-/// return type)].
+/// [ `group_key` (only for hash agg) / `sort_key` (only for max/mix without append only) /
```
Comment (BugenZhao) [inline src/stream/src/executor/aggregation/mod.rs:356]:
```suggestion
/// [ `group_key` (only for hash agg) / `sort_key` (only for extreme) /
```

**Review by BugenZhao — APPROVED** [review]
LGTM. BTW, which bug does this PR fix?

File: src/stream/src/executor/aggregation/mod.rs [hunk 2]
```diff
@@ -396,6 +397,37 @@ pub fn generate_column_descs(
...
+    // Always leave 1 space for agg call value.
+    let relational_pk_len = table_desc.len() - 1;
+    StateTable::new(
+        ks,
+        table_desc,
+        // Primary key do not includes group key.
+        vec![
+            // Now we only infer order type for min/max in a naive way.
+            if agg_call.kind == AggKind::Max {
+                OrderType::Descending
```
Comment (skyzh) [inline src/stream/src/executor/aggregation/mod.rs:421]: Would you please help check how we encode stream pk previously? For max agg, did we also encode pk in descending way?

**Review by skyzh — APPROVED** [review]

File: src/stream/src/executor/aggregation/mod.rs (hunk 1 above)
Comment (BowenXiao1999) [reply src/stream/src/executor/aggregation/mod.rs:356]: It is a fix when implementing #2999 . No direct case relates to it.

File: src/stream/src/executor/aggregation/mod.rs (hunk 2 above)
Comment (BowenXiao1999) [reply src/stream/src/executor/aggregation/mod.rs:421]: You mean write batch/keyspace? I think after this PR, as long as we are using state table, we are encode pk in correct order way.

**Pull request merged**